src/multilayerqg.jl

```diff
@@ -587,16 +623,50 @@ function pvfromstreamfunction!(qh, ψh, params::TwoLayerParams, grid)
   return nothing
 end
 
+"""
+    @kernel function streamfunctionfrompv_kernel!(ψh, qh, S⁻¹, nlayers)
+
+Kernel for GPU acceleration of streamfunction from PV calculation, i.e., invert the PV to obtain
+the Fourier transform of the streamfunction `ψh` in each layer from `qh` using `ψh = params.S⁻¹ qh`.
+"""
+@kernel function streamfunctionfrompv_kernel!(ψh, qh, S⁻¹, nlayers)
+    i, j = @index(Global, NTuple)
+
+    @unroll for k = 1:nlayers
+
+        @inbounds ψh[i, j, k] = 0
+
+        @unroll for m = 1:nlayers
+            @inbounds ψh[i, j, k] += S⁻¹[i, j][k, m] * qh[i, j, m]
+        end
+    
+    end
+end
+
```

Comment:
are the two kernel functions identical except the order they want their arguments?

Reply:
Looks like it yes. It'd probably make sense to write one kernel in the more general form.

Reply:
I think so! Would make the code more robust.